### PR TITLE
Fix of LogsCheckingAfterRestartTest

### PR DESCRIPTION
**Pull request by Mykhailenko**
Workaround to fix LogsCheckingAfterRestartTest test. In this test we start, stop, and start server again.
When we perform second start, during initializing process RMCore retrieves LocalNodes nodesource from db. And after initialization we try to add another nodesource with the same name. So that is why I added isNodeSourceAlreadyExist method to RMCore, thus during server start-up we can check first if there is already nodesource with this name and if it exists we do not add new nodesource.

### CODE REVIEW

File: rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/common/util/RMProxyUserInterface.java [hunk 1]
```diff
@@ -366,6 +366,11 @@ public NodeSet getNodes(Criteria criteria) {
         return target.getNodes(criteria);
     }
 
+    @Override
+    public boolean isNodeSourceAlreadyExist(String nodeSourceName) {
```
Comment (marcocast) [inline rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/common/util/RMProxyUserInterface.java]: isNodeSourceAlreadyExisting

File: rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/common/util/RMProxyUserInterface.java [hunk 2]
```diff
@@ -366,6 +366,11 @@ public NodeSet getNodes(Criteria criteria) {
...
+        return this.target.isNodeSourceAlreadyExist(nodeSourceName);
```
Comment (marcocast) [inline rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/common/util/RMProxyUserInterface.java]: isNodeSourceAlreadyExist

File: rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/frontend/ResourceManager.java [hunk 3]
```diff
@@ -350,6 +350,13 @@ NodeSet getNodes(int number, TopologyDescriptor descriptor, List<SelectionScript
      */
     NodeSet getNodes(Criteria criteria);
 
+    /**
+     *
+     * @param nodeSourceName
+     * @return true if RM already has node source with given name
+     */
+    boolean isNodeSourceAlreadyExisting(String nodeSourceName);
```
Comment (fviale) [inline rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/frontend/ResourceManager.java]: Better to use BooleanWrapper to let the caller decide to handle it synchronously or not.

Comment (Mykhailenko) [reply rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/frontend/ResourceManager.java]: yes right, make sense

**Pull request closed without merging**